galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemCicsImpl.java

```diff
@@ -78,42 +82,75 @@ public SemCicsImpl(SemManagerImpl semManager, ICicstsManagerSpi cicstsManager, S
         if (this.complexCics == null) {
             throw new SemManagerException("Unable to locate CICS Region in SEM complex with applid '" + getApplid() + "'");
         }
-
+        
         try {
+        	
+        	// Get a runtime job for this CICS
             ArrayList<Job> jobs = new ArrayList<>();
             this.complexCics.Build_Runtime_Jobs(this.complex, jobs);
 
+            // Only one job should found
             if (jobs.size() != 1) {
                 throw new SemManagerException("More than one runtime JCL returned from complex");
             }
 
+            // Get the single job in the array
             Job job = jobs.get(0);
+            
+            // Set the constant information of this CICS job
             this.jobname = job.getJobname();
             this.vtamnode = this.complexCics.getVtamnode();
-
-            String runtimeJcl = semZosHandler.generateJCL(jobs.get(0));
-
-            StringBuilder sb = new StringBuilder();
-            sb.append("/*JOBPARM SYSAFF=");
-            sb.append(getZosImage().getSysname());
-            sb.append("\n");
-
-
-            if (this.vtamnode != null && vtamnode.length() > 0) {
-                sb.append("//NET      COMMAND 'VARY NET,ID=");
-                sb.append(this.vtamnode);
-                sb.append(",ACT'\n");
-            }
-
-            sb.append(runtimeJcl);
-
-            this.jcl = sb.toString();
-
-        } catch(Exception e) {
-            throw new SemManagerException("Unable to generate CICS region JCL",e);
-        }
+            
+            // Build JCL to startup CICS
+            buildCicsJcl(job);
+            
+        } catch (Exception e) {
+        	
+            throw new SemManagerException("Could not generate the CICS runtime JCL", e);
+        }    
     }
 
+    /**
+     * From a batch job, generate some JCL to startup CICS
+     * and activate the APPID of the CICS job. Store the JCL
+     * ready for CICS startup.
+     * 
+     * @param job
+     * @throws SemManagerException
+     */
+	private void buildCicsJcl(Job job) throws SemManagerException {
+
+		try {
+			
+			// Build some skeleton JCL
+			String runtimeJcl = semZosHandler.generateJCL(job);
+
+			StringBuilder sb = new StringBuilder();
+			
+			// Append the start of the JCL with system information
+			sb.append("/*JOBPARM SYSAFF=");
+			sb.append(getZosImage().getSysname());
+			sb.append("\n");
+
+			// Append the JCL with commands to initialise the APPLID
+			if (this.vtamnode != null && vtamnode.length() > 0) {
+				sb.append("//NET      COMMAND 'VARY NET,ID=");
+				sb.append(this.vtamnode);
+				sb.append(",ACT'\n");
+			}
+
+			// Append the runtime CICS JCL to existing JCL
+			sb.append(runtimeJcl);
+
+			// Store the JCL ready for runtime
+			this.jcl = sb.toString();
+			
+		} catch (Exception e) {
+			
+            throw new SemManagerException("Unable to generate CICS region JCL", e);
+		}
```

Comment:
can we isolate this try catch to just the line of code that could throw an exception if possible